arch/lkl/kernel/signal.c

```diff
@@ -50,25 +61,100 @@ void lkl_process_trap(int signr, struct ucontext *uctx)
         /* Handle required signal */
         if(signr == ksig.sig)
         {
+            LKL_TRACE("trap task %p %d\n", current, signr);
             handle_signal(&ksig, uctx);
             break;
         }
     }
 }
 
-void do_signal(struct pt_regs *regs)
+/*
+    Walk to the end of the list and make it point at the new node
+*/
+
+
+void append_ksignal_node(struct thread_info *task, struct ksignal_list_node* node)
 {
-    struct ksignal ksig;
-    struct ucontext uc;
+    volatile struct ksignal_list_node **node_ptr;
+    spin_lock(&task->signal_list_lock);
+    node_ptr = &task->signal_list;
 
-    LKL_TRACE("enter\n");
+    while (*node_ptr) {
+        volatile struct ksignal_list_node *next_node = *node_ptr;
+        node_ptr = &(next_node->next);
+    }
+        
+    *node_ptr = node;
+    spin_unlock(&task->signal_list_lock);
+}    
 
-    memset(&uc, 0, sizeof(uc));
-    initialize_uctx(&uc, regs);
+// probably needs to be called with the cpu lock
+void move_signals_to_task(void)
+{
+    struct ksignal ksig;    // place to hold retrieved signal
+    // get the lkl local version of the current task, so we can store the signals
+    // in a list hanging off it.
+    struct thread_info *current_task;
+    current_task = task_thread_info(current); 
     while (get_signal(&ksig)) {
-        LKL_TRACE("ksig.sig=%d", ksig.sig);
+<<<<<<< HEAD
```

Comment:
You appear to have a merge conflict that didn't get fully cleaned up

Reply:
If only typing make built it...